File: app/src/main/java/com/example/bindechexclock/ui/notifications/NotificationsViewModel.java
package com.example.bindechexclock.ui.notifications;

import androidx.lifecycle.LiveData; // Changed
import androidx.lifecycle.MutableLiveData; // Changed
import androidx.lifecycle.ViewModel; // Changed

public class NotificationsViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public NotificationsViewModel() {
        mText = new MutableLiveData<>();
        // mText.setValue("This is notifications fragment"); // Default text removed
    }

    public LiveData<String> getText() {
        return mText;
    }

    public void setmText(String s) {
        mText.setValue(s);
    }
}

File: app/src/main/java/com/example/bindechexclock/ui/notifications/NotificationsViewModel.kt
package com.example.bindechexclock.ui.notifications

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class NotificationsViewModel : ViewModel() {

    private val _text = MutableLiveData<String>()
    val text: LiveData<String> get() = _text

    init {
        // Default text, if any, would be set here.
        // For this app, the Fragment usually sets the initial text.
    }

    fun updateText(newText: String) {
        _text.value = newText
    }
}
